### PR TITLE
Cats version

### PR DESCRIPTION
**Pull request by mandubian**

### CODE REVIEW

**Comment by djspiewak** [comment]
Seems very outdated.  Doesn't appear to have any of the recent changes (or tests) to support higher arities (e.g. `Either`) or higher-order constructors (e.g. `Free`) or any of the special casing to support `State`.

**Comment by mandubian** [comment]
Yes, I wanted to recreate it on an another branch of mine to be able to resync with your master!
Moving to cats finally?

**Comment by djspiewak** [comment]
Go for it!  (in other words, just swap the Cats implementation in for the existing Scalaz one; no need to create a separate module)

**Comment by mandubian** [comment]
and what about providing shapeless support in a separate module to play with Miles tricks (because I believe Generic1 could help State)

**Comment by djspiewak** [comment]
Shapeless support in a separate module would be excellent.  I don't know how much `Generic1` will help `State` (aside from the obvious `corecurse1`/`corecurse2`/etc bloat), but I'm very willing to keep an open mind!

**Comment by mandubian** [comment]
Does your `Driver` work for all the cases you wanted or just for a few?

**Comment by djspiewak** [comment]
Oh, `Driver` works for basically everything.  The problem is the rest of the machinery.  I'm still trying to figure out how to do that.

To be clear, `State` is supported in master right now, but only as the outer-most effect.  `Driver` allows `State` to be an inner effect by changing the definition of `Effects` from a composition of effects to "some arbitrary dependently-typed blob", rewriting `State` into `StateT`.  The trick is making that work at the value level for anything more complicated than `point`.

**Comment by mandubian** [comment]
@djspiewak ok here we are! Full cats support, still using scalaz Task for tests (a new Task is being studied in Typelevel scope so we'll see later), I had to spend time on understanding the role of expand because I was fooled by the name thinking it was something else :)

File: src/test/scala/emm/EmmSpecs.scala [hunk 1]
```diff
@@ -176,7 +203,7 @@ object EmmSpecs extends Specification {
 
       "inner" >> {
         type E = Option |: Free[List, ?] |: Base
-
+        implicitly[Effects.Traverser[Free[List, ?] |: Base]]
```
Comment (djspiewak) [inline src/test/scala/emm/EmmSpecs.scala]: Forget to delete something?  :-)

Comment (mandubian) [inline src/test/scala/emm/EmmSpecs.scala]: I think so :) Let me delete it!

**Comment by mandubian** [comment]
Nice, I'll try to think about the State issue during weekend but isn't it just structural as you said? State is defined as refined StateT so...

**Comment by djspiewak** [comment]
It is basically structural. It gets tricky though when it comes down to actually implementing a fix. Basically, you can no longer assume a "onion like" structure to the value of `run`, which significantly complicates every function.  Still working through some ideas on that front.

**Pull request merged**